File: FancyRatProject_Part2/POO/Pets/Cat.java
public class Cat extends Pet {

    public Cat() {

    }

    public Cat(String name, String breed, String color) {
        super(name, breed, color, 0); // The age is set to 0 for cats; you can set the actual age as per your requirement.
    }

    @Override
    public void eat() {
        // Implement the eating behavior of the cat
        System.out.println("The cat is eating.");
    }

    @Override
    public void run() {
        // Implement the running behavior of the cat
        System.out.println("The cat is running.");
    }

    @Override
    public void sleep() {
        // Implement the sleeping behavior of the cat
        System.out.println("The cat is sleeping.");
    }

    @Override
    public String toString() {
        return "Cat: { " + super.toString() + " }"; 
    }
}

File: FancyRatProject_Part2/POO/Animals/Cat.py
from abc import ABC, abstractmethod

class Animal(ABC):
    @abstractmethod
    def eat(self):
        pass

    @abstractmethod
    def run(self):
        pass

    @abstractmethod
    def sleep(self):
        pass

    @abstractmethod
    def sound(self):
        pass

class Cat(Animal):
    def __init__(self, name, age, breed):
        self.name = name
        self.age = age
        self.breed = breed

    def sound(self):
        return "Meow"

    def run(self):
        return f"{self.name} is running.."
    
    def sleep(self):
        return f"{self.name} is sleeping grrrr..."
    
    def eat(self):
        return f"{self.name} is eating..."
    
# Example usage
cat = Cat("Whiskers", 4, "Persian")
print(cat.name)  # Output: Whiskers
print(cat.age)  # Output: 4
print(cat.breed)  # Output: Persian
print(cat.sound())  # Output: Meow
print(cat.eat()) # Output Whiskers is eating...
print(cat.run()) # Output: Whiskers is running..
print(cat.sleep()) # Output : Whiskers is sleeping grrrr...